GVRf/Framework/framework/src/main/java/org/gearvrf/GVRScene.java

```diff
@@ -445,13 +448,22 @@ public void bindShaders(GVRSceneObject root) {
      * @see GVRScene.getLightList
      */
     protected boolean addLight(GVRLightBase light) {
-        if (light != null) {
-            Integer lightIndex = mLightList.size();
-            String name = "light" + lightIndex.toString();
-            mLightList.add(light);
-            NativeLight.setLightID(light.getNative(), name);
-            NativeScene.addLight(getNative(), light.getNative());
-            return true;
+        if (light == null)
+        {
+        	return false;
+        }
+        Integer lightIndex = mLightList.size();
+            
+        if (lightIndex >= MAX_LIGHTS)
+        {
+        	return false;
```

Comment:
This would be an awkward place to have an exception because addSceneObject may indirectly call this function. I think I should emit an error event when that PR makes it in so the application can trap this error and modify their light usage.